gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionMap.java

```diff
@@ -1017,13 +1014,20 @@ public final RegionEntry initRecoveredEntry(Object key, DiskEntry.RecoveredEntry
         } // synchronized
       }
       if (owner != null) {
-        owner.updateSizeOnCreate(key, owner.calculateRegionEntryValueSize(newRe));
+        int newSize = owner.calculateRegionEntryValueSize(newRe);
+        owner.updateSizeOnCreate(key, newSize);
         if (newRe.isTombstone()) {
           // refresh the tombstone so it doesn't time out too soon
           owner.scheduleTombstone(newRe, newRe.getVersionStamp().asVersionTag());
         }
         
         incEntryCount(1); // we are creating an entry that was recovered from disk including tombstone
+
+        // If owner is a local region i.e during backup account for entry overhead and value here.
+        // If owner is null i.e. in the case of recovery accounting is done in PlaceHolderDiskRegion
+        if (!newRe.isTombstone()) {
+          owner.acquirePoolMemory(0, newSize, true, null, true);
```

Comment:
May be not relevant for this PR. But a more appropriate name of the method "acquirePoolMemory" seems to be "reduceAvailableMemory" because we are trying to acquire after the value is already sitting in the memory.

Reply:
Yes, I understand that. But nitpicking :-) ...  I hinted at having two functions ( both internally doing the same thing ) but the names could have been different so that they showed both the scenario clearly.